File: upwork-intelligence/persist-searches.mjs
#!/usr/bin/env node
import fs from 'fs';
import path from 'path';

const dir = path.dirname(new URL(import.meta.url).pathname);
const file = process.argv[2];
if (!file) {
  console.error('Usage: node persist-searches.mjs <batch.json>');
  process.exit(1);
}
const batch = JSON.parse(fs.readFileSync(file, 'utf8'));
const list = Array.isArray(batch) ? batch : batch.searches || [];
const out = path.join(dir, 'search-results.jsonl');
for (const entry of list) {
  if (!entry?.keyword) continue;
  fs.appendFileSync(out, JSON.stringify(entry) + '\n');
}
const batchDir = path.join(dir, 'raw_batches');
fs.mkdirSync(batchDir, { recursive: true });
fs.writeFileSync(path.join(batchDir, `batch-${Date.now()}.json`), JSON.stringify(list));
console.log(JSON.stringify({ appended: list.length, out }));

File: upwork-intelligence/persist-searches.py
#!/usr/bin/env python3
"""Persist [{keyword, group, jobs}] to raw_batches/. Usage: persist-searches.py FILE.json"""
import json
import re
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parent
BATCH = ROOT / "raw_batches"
BATCH.mkdir(exist_ok=True)

data = json.loads(Path(sys.argv[1]).read_text())
for item in data:
    kw = item["keyword"]
    grp = item["group"]
    jobs = item.get("jobs") or []
    slug = re.sub(r"[^a-z0-9]+", "-", kw.lower()).strip("-")[:80]
    (BATCH / f"{slug}.json").write_text(
        json.dumps({"keyword": kw, "group": grp, "jobs": jobs})
    )
print(len(data), "written")
